lib/rust/ensogl/core/src/system/gpu/shader/compiler.rs

```diff
@@ -414,17 +414,18 @@ impl CompilerData {
 
     #[profile(Debug)]
     fn poll_programs_ready(&mut self) {
-        let ready_jobs = self.jobs.link_poll.drain_filter(|job| {
+        let ready_jobs = self.jobs.link_poll.extract_if(|job| {
             job.khr.is_program_ready(&self.context, &job.program).unwrap_or(true)
         });
         self.jobs.link_check.extend(ready_jobs.map(|job| job.map_input(|t| t.program)));
     }
 
     #[profile(Debug)]
     fn poll_shaders_ready(&mut self) {
-        let ready_jobs = self.jobs.compile_poll.drain_filter(|job| {
-            job.khr.is_shader_ready(&self.context, &job.program).unwrap_or(true)
-        });
+        let ready_jobs = self
+            .jobs
+            .compile_poll
+            .extract_if(|job| job.khr.is_shader_ready(&self.context, &job.program).unwrap_or(true));
```

Comment:
another example of `extract_if` being much worse than `drain_filter` regarding code readability

Reply:
I don't understand your comment. It's literally change in name + rustfmt, and it reads perfectly fine ("ready jobs = extract by condition from compile_poll" instead of "ready jobs = drain filter compile poll by condition")